From c067876e587ac55cd139f054900ca5219089ae22 Mon Sep 17 00:00:00 2001
From: Mandell Degerness <mdegerne@gmail.com>
Date: Mon, 21 Nov 2011 23:54:36 +0000
Subject: [PATCH 81/94] Fix to correctly report memory on Linux 3.X.

If sys is compiled on a Linux 3.X system, it returns LINUX3 instead of
LINUX2 for sys.platform.upper().

Change-Id: I78dae37d4facbaf6513c0eab651b1a06b93949b2
---
 nova/virt/libvirt/connection.py |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/nova/virt/libvirt/connection.py b/nova/virt/libvirt/connection.py
index 36938bf..333d0cd 100644
--- a/nova/virt/libvirt/connection.py
+++ b/nova/virt/libvirt/connection.py
@@ -1348,7 +1348,7 @@ class LibvirtConnection(driver.ComputeDriver):
 
         """
 
-        if sys.platform.upper() != 'LINUX2':
+        if sys.platform.upper() not in ['LINUX2', 'LINUX3']:
             return 0
 
         meminfo = open('/proc/meminfo').read().split()
@@ -1389,7 +1389,7 @@ class LibvirtConnection(driver.ComputeDriver):
 
         """
 
-        if sys.platform.upper() != 'LINUX2':
+        if sys.platform.upper() not in ['LINUX2', 'LINUX3']:
             return 0
 
         m = open('/proc/meminfo').read().split()
-- 
1.7.6.5

